x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwkSetLoader.java

```diff
@@ -68,24 +72,22 @@ public JwkSetLoader(final RealmConfig realmConfig, List<String> allowedJwksAlgsP
 
         // Any exception during loading requires closing JwkSetLoader's HTTP client to avoid a thread pool leak
         try {
-            final PlainActionFuture<Tuple<Boolean, JwksAlgs>> future = new PlainActionFuture<>();
+            final PlainActionFuture<Void> future = new PlainActionFuture<>();
             reload(future);
             // ASSUME: Blocking read operations are OK during startup
-            final Boolean isUpdated = future.actionGet().v1();
-            assert isUpdated : "initial reload should have updated the JWK set";
-        } catch (Throwable t) {
+            future.actionGet(TimeValue.timeValueMinutes(1));
+        } catch (Exception e) {
+            logger.debug("Could not load JWT realm's JWK sets within [1] minute during startup. Will lazy retry if necessary.", e);
             close();
-            throw t;
```

Comment:
Couple comments:
* It is not really necessary to have the 1 minute timeout because underlying code should have its own timeout. For example, the HTTP client has [various 5 second timeouts](https://github.com/elastic/elasticsearch/blob/42fbe0b1dcc3dfce4e8976f618110b227078916b/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/jwt/JwtRealmSettings.java#L125-L127). We have similar code logic in other realms and I don't recall any report of node hanging on startup because of them. Did you observe that in your local tests?
* The logging level should be increased to at least to info level. It's an important message and we have similar logging at warn level in [SamlRealm](https://github.com/elastic/elasticsearch/blob/42fbe0b1dcc3dfce4e8976f618110b227078916b/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlRealm.java#L806-L807) 
* We still closes the HTTP client on exception. So if JWKS is from an URL, it cannot be reloaded later since the HTTP client is closed. This also suggests that we need a test for it, i.e. JWKS is reloadable after initial failure.

Reply:
> We still closes the HTTP client on exception. So if JWKS is from an URL, it cannot be reloaded later since the HTTP client is closed. 

Good catch, thanks !  For this reason I reverted the change to start up and will address in a follow up. 

> It is not really necessary to have the 1 minute timeout because underlying code should have its own timeout. 

I agree, it should not be necessary.  But we should have a general rule to avoid potentially blocking startup indefinitely especially due to an external resource.  We should also have a general rule to not catch (or throw) Errors. We should also have a general rule that recoverable exceptions won't prevent startup. 

> The logging level should be increased to at least to info level. It's an important message and we have similar logging at warn level in [SamlRealm](https://github.com/elastic/elasticsearch/blob/42fbe0b1dcc3dfce4e8976f618110b227078916b/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlRealm.java#L806-L807)

Not sure I agree,  it is back to where it was for now. But the reason i down graded it was that  I don't think there is anything useful a user would gain with that information (especially if doesn't cause the server to not start). It won't fail authentication, rather it will lazily retry when needed and best case scenerio it works with a tiny delay or it will fail authentication because the reload also fails. In both cases, IMO, the message is largely inconsequential.